test/integration/shoots/care/shoot_care.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+	Overview
+		- Tests the Care Controller in the Gardenlet
+
+	Prerequisite
+		- Shoot Cluster with  Condition "APIServerAvailable" equals true
+
+	Test: Scale down API Server deployment of the Shoot in the Seed
+	Expected Output
+		- Shoot Condition "APIServerAvailable" becomes unhealthy
+ **/
+
+package care
+
+import (
+	"context"
+	"time"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardencorev1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	testframework "github.com/gardener/gardener/test/framework"
+	"github.com/gardener/gardener/test/integration/framework"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+)
+
+const (
+	timeout = 600 * time.Second
+)
+
+var _ = ginkgo.Describe("Shoot Care testing", func() {
+
+	f := testframework.NewShootFramework(nil)
+
+	f.Default().Release().CIt("Should observe failed Health condition in the Shoot when scaling down the API Server of the Shoot", func(ctx context.Context) {
+		var (
+			origReplicas *int32
+			err          error
+		)
+
+		defer func(ctx context.Context) {
+			if origReplicas != nil {
+				f.Logger.Infof("Test cleanup: Scale API Server to '%d' replicas again", *origReplicas)
```

Comment:
`*origReplicas` can be `nil` if the test fails on the first assertion.

Reply:
Yes, thats why there is a nil check?

Reply:
Sure, missed it...